src/linnia.js

```diff
@@ -35,10 +36,20 @@ class Linnia {
     this._users = _util.truffleHack(_users);
     this._records = _util.truffleHack(_records);
     this._permissions = _util.truffleHack(_permissions);
-    // set linniaContractUpgradeHubAddress address
-    // using user defined address
-    this._hubAddress = opt.linniaContractUpgradeHubAddress;
-    this._tokenAddress = opt.linniaTokenContractAddress;
+
+    if(opt){
+      this._hubAddress = opt.hubAddress || undefined;
+      this._tokenAddress = opt.tokenAddress || undefined;
+    }
+
+    await web3.version.getNetwork((err, netId) => {
+      const network = (netId == 4)? 'rinkeby' : 'ropsten'
```

Comment:
We can't assume !== 'rinkby' === 'ropsten'. Best to directly match IDs to networks, and if the ID is not supported, don't set addresses